src/test/java/org/opensearch/geospatial/ip2geo/action/UpdateDatasourceIT.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import lombok.SneakyThrows;
+
+import org.opensearch.client.ResponseException;
+import org.opensearch.geospatial.GeospatialRestTestCase;
+import org.opensearch.geospatial.GeospatialTestHelper;
+import org.opensearch.geospatial.ip2geo.Ip2GeoDataServer;
+import org.opensearch.rest.RestStatus;
+
+public class UpdateDatasourceIT extends GeospatialRestTestCase {
+    // Use this value in resource name to avoid name conflict among tests
+    private static final String PREFIX = UpdateDatasourceIT.class.getSimpleName().toLowerCase(Locale.ROOT);
+
+    @SneakyThrows
+    public void testUpdateDatasource_whenValidInput_thenUpdated() {
+        Ip2GeoDataServer.start();
```

Comment:
How about use Before and After annotations to handle the start and stop of Ip2GeoDataServer for all the test cases?